app/repositories/IndividualCheckRepository.scala

```diff
@@ -17,105 +17,76 @@
-import play.api.libs.json.{JsObject, Json}
-import play.modules.reactivemongo.ReactiveMongoApi
-import reactivemongo.api.WriteConcern
-import reactivemongo.play.json.collection.Helpers.idWrites
-import reactivemongo.play.json.collection.JSONCollection
-import reactivemongo.api.indexes.IndexType
+import models.{BinaryResult, IndividualCheckCount, MongoDateTimeFormats, OperationFailed, OperationSucceeded}
+import play.api.libs.json.Json
+import org.mongodb.scala.model._
+
+import java.util.concurrent.TimeUnit
 import play.api.Configuration
-import java.time.LocalDateTime
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.{Codecs, PlayMongoRepository}
 
+import java.time.LocalDateTime
 import scala.concurrent.{ExecutionContext, Future}
 
 // Tested in integration testing
 // $COVERAGE-OFF$
-class IndividualCheckRepository @Inject()(mongo: ReactiveMongoApi, config: Configuration)(implicit ec: ExecutionContext) {
-
-  val collectionName     : String = "individual-check-counters"
-
-  private val expireAfterSeconds = config.get[Int]("mongodb.ttl")
-
-  private val lastUpdatedIndex = MongoIndex(
-    key = Seq("lastUpdated" -> IndexType.Ascending),
-    name = "last-updated-index",
-    expireAfterSeconds = Some(expireAfterSeconds),
-    unique = false
-  )
-
-  private def collection : Future[JSONCollection] = for {
-      _ <- ensureIndexes
-      res <- mongo.database.map(_.collection[JSONCollection](collectionName))
-    } yield res
-
-  private lazy val idIndex = MongoIndex(
-    key = Seq("id" -> IndexType.Ascending),
-    name = "id-index",
-    unique = true
-  )
-
-  private def ensureIndexes = {
-    for {
-      collection                           <- mongo.database.map(_.collection[JSONCollection](collectionName))
-      createdLastUpdatedIndex              <- collection.indexesManager.ensure(lastUpdatedIndex)
-      createdIdIndex                       <- collection.indexesManager.ensure(idIndex)
-
-    } yield createdLastUpdatedIndex && createdIdIndex
-  }
+class IndividualCheckRepository @Inject()(mongo: MongoComponent, config: Configuration)(implicit ec: ExecutionContext)
+  extends PlayMongoRepository[IndividualCheckCount] (
+    collectionName = "individual-check-counters",
+    mongoComponent = mongo,
+    domainFormat = IndividualCheckCount.format,
+    indexes = Seq(
+      IndexModel(
+        Indexes.ascending("lastUpdated"),
+        IndexOptions()
+          .name("last-updated-index")
+          .expireAfter(config.get[Int]("mongodb.ttl"), TimeUnit.SECONDS)
+          .unique(false)
+      ),
+      IndexModel(
+        Indexes.ascending("id"),
+        IndexOptions()
+          .name("id-index")
+          .unique(true)
+      )
+    )
+  ){
 
   def getCounter(id: String): Future[Int] = {
-    val selector = Json.obj("id" -> Json.toJson(id))
+    val selector = Filters.equal("id", id)
+    val res = collection.find(selector).headOption()
```

Comment:
looking at the mongo migration guide we should be using the toFutureOption at the end, https://github.com/hmrc/hmrc-mongo/blob/main/MIGRATION.md#update-queries-to-new-syntax

Reply:
The alternative would be doing '.first().toFutureOption()' but I wasn't sure if this would be cleaner over just doing a .headOption() since they both give the same type Future[Option[IndividualCheckCount]]

Reply:
Changed to use - `val res = collection.find(selector).first().toFutureOption()`

- [Code Changes](https://github.com/hmrc/trusts-individual-check/pull/40/files#diff-3897281f4c0824f2e3cfb1e54d5f8a28088510d9ea60c77a7ec7461bbd290999R59)

Reply:
The only thing I can think of is maybe doing the .first() could throw some errors if nothing is returned and it might be safer doing a .headOption()? Though i'm not 100% sure, i'd have to check how .first() functions